.pre-commit-config.yaml

```
@@ -1,25 +1,48 @@
+default_stages: [pre-commit, pre-push]
+default_install_hook_types: [pre-commit, pre-push]
 repos:
 - repo: https://github.com/gitleaks/gitleaks
   rev: v8.21.1
   hooks:
   - id: gitleaks
+    stages: [pre-push]
```

Can be a follow up but should there actually be a .gitleaks.toml for this? I was able to pass a couple of "secrets" I while testing though I could have been doing it wrong.

When I made a .gitleaks.toml, I added some content like
```
title = "gitleaks config"

[[rules]]
description = "AWS Access Key"
regex = '''AWS_SECRET_ACCESS_KEY\s*=\s*['"]?([A-Z0-9]{20})['"]?'''
```

With this, the gitleaks hook actually triggered.

---

Good follow up tbh. 
1. We can create and add `.gitleaks.toml` to .gitignore
2. Have `.gitleaks.toml` with your custom `AWS_SECRET_ACCESS_KEY` rule added to it. 
3. Also broadcast `.gitleaks.toml` to the community on slack channel to add their own rules. 

Please feel free to lead this effort / create follow up issues on this.

---

```suggestion
    stages: [pre-commit]
```
Moreover, thinking about gitleaks running at `pre-push` stage; I would rather have gitleaks hook run at every commit than at pre-push stage. 

It be better to block commit a on gitleaks failure than the dev going to each and every commit to change it in the worst case scenario.

---

Sounds good. This has been updated.